buildpack/bp_descriptor.go

```diff
@@ -38,6 +63,19 @@ func ReadBpDescriptor(path string) (*BpDescriptor, error) {
 	if descriptor.WithRootDir, err = filepath.Abs(filepath.Dir(path)); err != nil {
 		return &BpDescriptor{}, err
 	}
+
+	apiVersion, err := api.NewVersion(descriptor.WithAPI)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(descriptor.Targets) == 0 && apiVersion.AtLeast("0.12") {
```

Comment:
```suggestion
	if len(descriptor.Targets) == 0 {
```
I apologize, the lifecycle issue and spec PR were not clear about this - we actually want to infer the target data for all (and especially older) Buildpack API; this is part of our [backwards compatibility strategy](https://github.com/buildpacks/spec/pull/336#issuecomment-1454099934). The only time we want to guard on API versions is when we write data.